README.md

```diff
@@ -4,60 +4,583 @@ A test-framework-agnostic set of helpers for testing Ember.js applications.
+
+Returns **[Promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise)&lt;void>** resolves when settled
+
+#### validateErrorHandler
+
+Validate the provided error handler to confirm that it properly re-throws
+errors when `Ember.testing` is true.
+
+This is intended to be used by test framework hosts (or other libraries) to
+ensure that `Ember.onerror` is properly configured. Without a check like
+this, `Ember.onerror` could _easily_ swallow all errors and make it _seem_
+like everything is just fine (and have green tests) when in reality
+everything is on fire...
+
+**Parameters**
+
+-   `callback` **[Function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** the callback to validate (optional, default `Ember.onerror`)
+
+**Examples**
+
+_Example implementation for `ember-qunit`_
+
+```javascript
+import { validateErrorHandler } from '@ember/test-helpers';
+
+test('Ember.onerror is functioning properly', function(assert) {
+  let result = validateErrorHandler();
+  assert.ok(result.isValid, result.message);
+});
+```
 
-A quick summary of the exports from the `ember-test-helpers` module:
-
-* `setApplication` - This function is used to allow the rest of the setup
-  functions to build a valid container/registry which can be used to lookup
-  instances and factories from your application.
-* `setResolver` - When `setApplication` has not been ran, this function is used to allow the other
-  functions build a valid container/registry that is able to look objects up
-  from your application (just as a running Ember application would).
-* `setContext` - Invoked by the host testing framework to set the current testing context (generally
-  the `this` within a running test).
-* `getContext` - Used to retrieve the current testing context.
-* `unsetContext` - Used to ensure that all handles on the testing context are released (allowing GC).
-* `setupContext` - Sets up a given testing context with `owner`, `get`, `set`, etc properties.
-* `teardownContext` - Cleans up any objects created as part of the owner created in `setupContext`.
-* `setupRenderingContext` - Sets up the provided context with the ability to render template
-  snippets by adding `render`, `clearRender`, etc.
-* `teardownRenderingContext` - Cleans up any work done in a rendering test.
-* `settled` - Returns a promise which will resolve when all async from AJAX, test waiters, and
-  scheduled timers have completed.
-* `validateErrorHandler` - Used to ensure that the `Ember.onerror` error handler properly re-throws any errors during testing.
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** object with `isValid` and `message`
 
 ## Collaborating
 
 ### Installation
 
-* `git clone <repository-url>` this repository
-* `cd ember-test-helpers`
-* `yarn install`
+-   `git clone <repository-url>` this repository
+-   `cd ember-test-helpers`
+-   `yarn install`
```

Comment:
IMHO it's quite annoying that documentation is reformatting the whole file when it's only supposed to replace a single section. I've had the same issue in qunit-dom and decided to have it output in a API.md instead

Reply:
Ya, agreed that its odd.